warehouse/templates/accounts/login.html

```diff
@@ -45,58 +45,11 @@ <h1 class="page-title">{% trans title=title %}Log in to {{ title }}{% endtrans %
         {% endif %}
 
         <div class="form-group">
-          <label for="username" class="form-group__label">
-            {% trans %}Username{% endtrans %}
-            {% if form.username.flags.required %}
-            <span class="form-group__required">{% trans %}(required){% endtrans %}</span>
-            {% endif %}
-          </label>
-          {{ form.username(
-              placeholder=gettext("Your username"),
-              autocapitalize="off",
-              autocomplete="username",
-              spellcheck="false",
-              required="required",
-              class_="form-group__field",
-              aria_describedby="username-errors",
-          ) }}
-          <div id="username-errors">
-            {% if form.username.errors %}
-            <ul class="form-errors" role="alert">
-              {% for error in form.username.errors %}
-              <li>{{ error }}</li>
-              {% endfor %}
-            </ul>
-            {% endif %}
-          </div>
+          {{ input_field(name="username", user_friendly_name="Username", form_instance=form.username, placeholder="Your username", autocomplete="username") }}
```

Comment:
Thanks for the detailed feedback, I'll go back and do some fixing!

Reply:
Having trouble trying to figure out how to not pass "data_action" and "data_target" to "form_instance()" when they are not passed to the macro. Is there a simple way to not pass it if not set, or would I need to use an if statement in the macro?

Reply:
I have been busy recently, but will begin working on this again soon!

Reply:
Hey @AshtonWooster ! Any chance you've found some time to work on this again?

Reply:
@miketheman I should have some time this weekend to work on it, I'll let you know what I get done.

Reply:
@miketheman I was unsure how to regenerate the messages.pot file, or add the "data-*-target" and "data-*-action" for the macros. Are there any resources I can read through that explain how to do this? Thanks for the help!

Reply:
Certainly! For the translations, the dev docs discuss briefly: https://warehouse.pypa.io/development/submitting-patches.html#translations

This often has to be done when rebasing a branch, since the `main` branch gets updated with other strings, so if a PR falls behind, it has to be done before merging.

For creating `data-*-target` style attributes conditionally, I'm wondering if constructing those conditionally as a dictionary and passing them to the field would "just work", something like:

```jinja2
set extra_attributes = {}
if foo:
    extra_attributes["data-foo-action"] = "something"
...

input_field(
    ...,
	**extra_attributes,
)
```

or something like that? I'm not totally certain if that works, but worth a try!

Reply:
Thanks for the help, I'll try to implement this when I get a chance.